tools/automatic_query_fixer/src/test/java/com/google/cloud/bigquery/utils/queryfixer/BigQueryServiceTest.java

```diff
@@ -42,11 +54,85 @@ public void dryRun1() {
   }
 
   @Test
-  public void dryRun2() {
-    BigQueryService service = getService();
+  public void dryRun_tableNotFound() {
     String query = "SELECT corpus FROM `bigquery-public-data.samples.shakespearex`";
     BigQueryException exception = service.catchExceptionFromDryRun(query);
     assertNotNull(exception);
     assertEquals("Not found: Table", exception.getMessage().substring(0, 16));
   }
+
+  @Test
+  public void listTables() {
+    String project = "bigquery-public-data";
+    String dataset = "austin_311";
+    List<String> tables = service.listTableNames(project, dataset);
+    assertEquals(2, tables.size());
+    assertThat(tables, contains("311_request", "311_service_requests"));
+  }
+
+  /**
+   * Mock the connection to the BigQuery server. Use reflection to inject the fake connection into
+   * the {@link BigQueryService}.
+   */
+  private void fakeBigQuery() {
+    BigQuery bigQuery = mock(BigQuery.class);
```

Comment:
Since the service is going to be shared, we should init the variable before all tests starts and use mock annotation.

Reply:
use "@Mock" instead of mock()

Reply:
Here may not be a good place to use `@mock`. BigQuery object will be injected into the `BigQueryService` field, which will be shared by different tests. The BigQuery object will be only mocked once in the beginning, because it is only called by the `@Before` method.